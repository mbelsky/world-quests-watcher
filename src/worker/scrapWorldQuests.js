const urlTemplate = `https://www.wowhead.com/world-quests/{addon}/{region}`;

const addons = ["legion", "bfa"];
const regions = ["eu", "na"];

export async function scrapWorldQuests(scrapFn) {
  const htmlMap = {};

  for (const region of regions) {
    for (const addon of addons) {
      const url = urlTemplate
        .replace(/{addon}/, addon)
        .replace(/{region}/, region);

      const html = await scrapFn(url);

      htmlMap[`${region}-${addon}`] = html;
    }
  }

  return htmlMap;
}
