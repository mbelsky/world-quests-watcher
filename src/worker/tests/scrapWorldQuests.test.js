import { jest } from "@jest/globals";

import { scrapWorldQuests } from "../scrapWorldQuests.js";

test("urls creation", async () => {
  const scrap = jest.fn(() => Promise.resolve());

  await scrapWorldQuests(scrap);

  expect(scrap).toHaveBeenCalledTimes(4);

  expect(scrap).toHaveBeenNthCalledWith(1, "https://www.wowhead.com/world-quests/legion/eu");
  expect(scrap).toHaveBeenNthCalledWith(2, "https://www.wowhead.com/world-quests/bfa/eu");
  expect(scrap).toHaveBeenNthCalledWith(3, "https://www.wowhead.com/world-quests/legion/na");
  expect(scrap).toHaveBeenNthCalledWith(4, "https://www.wowhead.com/world-quests/bfa/na");
});

test("html mapper", async () => {
  const scrap = jest.fn((url) => Promise.resolve(url));

  const result = await scrapWorldQuests(scrap);

  expect(result).toMatchSnapshot()
})
