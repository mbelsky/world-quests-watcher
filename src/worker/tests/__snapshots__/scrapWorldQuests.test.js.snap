// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html mapper 1`] = `
Object {
  "eu-bfa": "https://www.wowhead.com/world-quests/bfa/eu",
  "eu-legion": "https://www.wowhead.com/world-quests/legion/eu",
  "na-bfa": "https://www.wowhead.com/world-quests/bfa/na",
  "na-legion": "https://www.wowhead.com/world-quests/legion/na",
}
`;
